compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrDeclarationStorage.kt

```diff
@@ -59,7 +59,7 @@ class Fir2IrDeclarationStorage(
     private val moduleDescriptor: FirModuleDescriptor
 ) : Fir2IrComponents by components {
 
-    internal var annotationGenerator: AnnotationGenerator? = null
+    private val annotationGenerator = AnnotationGenerator(this)
```

Comment:
I think it's better to add `annotationGenerator` to components

Reply:
Indeed, it is created five different places. :o Added it to `Fir2IrComponents`. I also grouped elements in the components into some relevant pieces; feel free to mingle them.